google-cloud-resourcemanager/src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java

```diff
@@ -16,58 +16,168 @@
+  private static final RetrySettings CREATE_RETRY_SETTINGS =
+      RetrySettings.newBuilder()
+          // SLO permits 30s at 90th percentile, double it for total limit
+          .setTotalTimeout(Duration.ofMinutes(1))
+          .setInitialRetryDelay(Duration.ofSeconds(5))
+          .setRetryDelayMultiplier(1.5)
+          // Observed P95 latency is 11s. We probably shouldn't sleep longer than this.
+          .setMaxRetryDelay(Duration.ofSeconds(11))
+          .setJittered(false)
+          .setInitialRpcTimeout(Duration.ofSeconds(5))
+          .setMaxRpcTimeout(Duration.ofSeconds(5))
+          .build();
+
+  // reference: https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+  private static final ImmutableMap<Integer, Integer> RPC_TO_HTTP_CODES =
+      ImmutableMap.<Integer, Integer>builder()
+          .put(0, 200)
+          .put(1, 499)
+          .put(2, 500)
+          .put(3, 400)
+          .put(4, 504)
+          .put(5, 404)
+          .put(6, 409)
+          .put(7, 403)
+          .put(16, 401)
+          .put(8, 429)
+          .put(9, 400)
+          .put(10, 409)
+          .put(11, 400)
+          .put(12, 501)
+          .put(13, 500)
+          .put(14, 503)
+          .put(15, 500)
+          .build();
+
+  private static final ResultRetryAlgorithm<Operation> OPERATION_HANDLER =
+      new ResultRetryAlgorithm<Operation>() {
+        @Override
+        public TimedAttemptSettings createNextAttempt(
+            Throwable prevThrowable, Operation prevResponse, TimedAttemptSettings prevSettings) {
+          return null;
+        }
+
+        @Override
+        public boolean shouldRetry(Throwable prevThrowable, Operation prevOp)
+            throws CancellationException {
+          return prevThrowable != null || prevOp.getDone() == null || !prevOp.getDone();
+        }
+      };
+
+  private final CloudResourceManager resourceManager;
+  private final ApiClock clock;
 
   public HttpResourceManagerRpc(ResourceManagerOptions options) {
     HttpTransportOptions transportOptions = (HttpTransportOptions) options.getTransportOptions();
     HttpTransport transport = transportOptions.getHttpTransportFactory().create();
     HttpRequestInitializer initializer = transportOptions.getHttpRequestInitializer(options);
     resourceManager =
-        new Cloudresourcemanager.Builder(transport, new JacksonFactory(), initializer)
+        new CloudResourceManager.Builder(transport, new JacksonFactory(), initializer)
             .setRootUrl(options.getHost())
             .setApplicationName(options.getApplicationName())
             .build();
+    clock = options.getClock();
   }
 
   private static ResourceManagerException translate(IOException exception) {
     return new ResourceManagerException(exception);
   }
 
+  private static ResourceManagerException translate(Status status) {
+    int code;
+    if (RPC_TO_HTTP_CODES.containsKey(status.getCode())) {
```

Comment:
This can be:

```java
Integer code = RPC_TO_HTTP_CODES.get(status.getCode());
if (code == null) {
    code = BaseHttpServiceException.UNKNOWN_CODE;
}
```
which is a typical technique, and is technically faster (one call to map instead of 2) and shorter.

Reply:
Done